addons/website_sale/controllers/main.py

```diff
@@ -1893,3 +1893,17 @@ def products_recently_viewed_delete(self, product_id, **kwargs):
         if visitor_sudo:
             request.env['website.track'].sudo().search([('visitor_id', '=', visitor_sudo.id), ('product_id', '=', product_id)]).unlink()
         return {}
+
+    # --------------------------------------------------------------------------
+    # Override dynamic filter to add a show variant option
+    # --------------------------------------------------------------------------
+    @route('/website/snippet/filters', type='json', auth='public', website=True)
+    def get_dynamic_filter(
+        self, filter_id, template_key, limit=None, search_domain=None, with_sample=False,
+        **kwargs,
+    ):
+        context = kwargs.get('context', {})
```

Comment:
And why can't this `with_context` be added directly in `website`? :thinking: 
Previous version was speaking about variants in `website`, that's why I said it shouldn't be there, but adding a context is OK, IMHO, as it's neutral and not "sale related".
Keep in mind that I'm no expert in `website_sale`, and even less about snippets, only lending a hand in the reviews, so I might overlook many things here, and just saying what sound strange to me, doesn't mean it's wrong though =].
Right now, the problem arises once `website` might add things to the original route, what would happen then?